### PR TITLE
Dockerstatic: Move gnupg and curl to top of deb11 dockerfile

### PR DESCRIPTION
**Pull request by Haroon-Khel**
- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

The dockerfile needs curl and gnupg at the beginning of the dockerfile for the curl and gnupg commands later in the image build

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 FROM debian:bullseye
 # Install Base Requirements
 RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
-RUN apt-get update && apt-get install -y perl openssh-server unzip zip wget apt-utils
+RUN apt-get update && apt-get install -y perl openssh-server unzip zip wget apt-utils gnupg curl
 RUN echo "y" | ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -P ""
```
Comment (zdtsw) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11:5]: isnt that duplicated as in https://github.com/adoptium/infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11#L36?

Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11:5]: i mean, why not move line 36 up to line 4

Comment (Haroon-Khel) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11:5]: Yeah, I forgot to remove them from later in the dockerfile (removed now). We need them to be installed earlier in the dockerfile so the curl and gpg commands can run

**Review by zdtsw — APPROVED** [review]

**Comment by karianna** [comment]
I'm missing the 'why' these are being added :-), is this for parity with other Dockerfiles?

**Comment by sxa** [comment]
> I'm missing the 'why' these are being added :-), is this for parity with other Dockerfiles?

+1 to this comment - despite the PR title it looks like the installation of these is being MOVED rather than added within the file. Please update the PR description with an explanation of the change.

**Comment by Haroon-Khel** [comment]
@sxa @karianna gnupg and curl need to be installed earlier in the dockerfile for the lines 
https://github.com/adoptium/infrastructure/blob/e50bb567927562dc7cfcca71d4c7fe81baf08e83/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11#L12 
and 
https://github.com/adoptium/infrastructure/blob/e50bb567927562dc7cfcca71d4c7fe81baf08e83/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/DockerStatic/Dockerfiles/Dockerfile.deb11#L13
to run without failure

**Pull request merged**